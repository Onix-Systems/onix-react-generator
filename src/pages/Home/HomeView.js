import React from 'react';
import PropTypes from 'prop-types';
import Button from '../../components/Button/Button';
import Loader from '../../components/Loader/Loader';
import styles from './sass/Home.module.scss';

export default function HomeView({
  information, getJoke, title, buttonTitle
}) {
  return (
    <div>
      <p className={styles.appIntro}>
        {title}
      </p>
      {information ? <p>{information}</p> : <Loader />}
      <Button
        title={buttonTitle}
        buttonClass={styles.updateJoke}
        handleClick={getJoke}
      />
    </div>
  );
}

HomeView.propTypes = {
  information: PropTypes.string.isRequired,
  getJoke: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  buttonTitle: PropTypes.string.isRequired
};
