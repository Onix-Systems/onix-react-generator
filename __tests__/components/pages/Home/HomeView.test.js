import React from 'react';
import userEvent from '@testing-library/user-event';
import i18next from 'i18next';
import { render } from '../../../settings/test-utils';
import HomeView from '../../../../src/pages/Home/HomeView';

describe('<HomeView />', () => {
  let wrapper;
  const mockInformation = 'mock information';
  const mockGetJoke = jest.fn();

  beforeEach(() => {
    wrapper = render(
      <HomeView
        information={mockInformation}
        getJoke={mockGetJoke}
        title={i18next.t('homePage')}
        buttonTitle={i18next.t('newJoke')}
      />
    );
  });

  test('should render HomeView component', () => {
    const { asFragment } = wrapper;
    expect(asFragment(wrapper)).toMatchSnapshot();
  });

  test('should render Button component', () => {
    const { getByTestId } = wrapper;
    const Button = getByTestId('button');
    expect(Button).toBeInTheDocument();
  });

  test('on Button click should call mockGetJoke function', () => {
    const { getByTestId } = wrapper;
    const Button = getByTestId('button');
    userEvent.click(Button);
    expect(mockGetJoke).toHaveBeenCalled();
  });
});
