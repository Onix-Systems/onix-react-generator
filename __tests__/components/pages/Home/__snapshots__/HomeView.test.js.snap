// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeView /> should render HomeView component 1`] = `
<DocumentFragment>
  <div>
    <p
      class="appIntro"
    >
      Home page
    </p>
    <p>
      mock information
    </p>
    <button
      class="updateJoke"
      data-testid="button"
      type="button"
    >
      New joke
    </button>
  </div>
</DocumentFragment>
`;
